exchanges/btcmarkets/btcmarkets.go

```diff
@@ -53,316 +84,555 @@ type BTCMarkets struct {
+		var strTemp strings.Builder
+		for x := range marketIDs {
+			strTemp.WriteString("marketId=" + marketIDs[x] + "&")
 		}
+		req["marketId"] = strTemp.String()[:strTemp.Len()-1]
 	}
-
-	return resp.Orders, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders, req, &resp)
 }
 
-// GetOrderDetail returns order information an a specific order
-// orderID - example "1337"
-func (b *BTCMarkets) GetOrderDetail(orderID []int64) ([]Order, error) {
-	type OrderDetail struct {
-		OrderIDs []int64 `json:"orderIds"`
-	}
-	orders := OrderDetail{}
-	orders.OrderIDs = append(orders.OrderIDs, orderID...)
+// FetchOrder finds order based on the provided id
+func (b *BTCMarkets) FetchOrder(id string) (OrderData, error) {
+	var resp OrderData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	resp := Response{}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodPost, btcMarketsOrderDetail, orders, &resp)
-	if err != nil {
-		return nil, err
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals(before, after, limit int64) ([]TransferData, error) {
+	var resp []TransferData
+	params := url.Values{}
+	if (before > 0) && (after >= 0) {
+		return resp, errors.New("BTCMarkets only supports either before or after, not both")
 	}
-
-	if !resp.Success {
-		return nil, errors.New(resp.ErrorMessage)
+	if before > 0 {
+		params.Set("before", strconv.FormatInt(before, 10))
+	}
+	if after >= 0 {
+		params.Set("after", strconv.FormatInt(after, 10))
 	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsWithdrawals, params),
+		nil,
+		&resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetWithdrawal gets withdrawawl info for a given id
+func (b *BTCMarkets) GetWithdrawal(id string) (TransferData, error) {
+	var resp TransferData
+	if id == "" {
+		return resp, errors.New("id cannot be an empty string")
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsWithdrawals+"/"+id,
+		nil, &resp)
+}
 
-		for x := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[x].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Volume /= common.SatoshisPerBTC
-		}
+// ListDeposits lists the deposit history
+func (b *BTCMarkets) ListDeposits(before, after, limit int64) ([]TransferData, error) {
+	var resp []TransferData
+	params := url.Values{}
+	if (before > 0) && (after >= 0) {
```

Comment:
Please move this check to the top, no need to set params or even create resp before